scripts/package.ts

```diff
@@ -171,10 +200,40 @@ function main() {
         }
 
         fs.writeFileSync(packageJsonFile, JSON.stringify(packageJson, undefined, '    '))
-        child_process.execFileSync('vsce', ['package', '--ignoreFile', '../.vscodeignore.packages'], {
-            stdio: 'inherit',
-            shell: process.platform === 'win32', // For vsce.cmd on Windows.
-        })
+
+        // vsce runs this command by default.
+        // This can result in errors that originate from configuration in the src.gen/ directory, but that directory
+        // cannot be managed directly since it is autogenerated code.
+        // This section runs the validation command manually and ignores errors only related to src.gen/
```

Comment:
Is this logic intended to actually tell us if dependabot is trying to recurse into src.gen? Or is it still expected that it will?

Reply:
Its unrelated to dependabot. We are basically just replacing an internal validation step of `vsce package` with our own that will not throw an error on issues with `src.gen/`. Without this, splitting off `src.gen` from the root repo results in a dependency error during packaging based on something in `src.gen/node_modules`. The extension appears to work normally with this dependency error:
```
npm ERR! invalid: typescript@5.7.0-dev.20240926 /Volumes/workplace/aws-toolkit-vscode/src.gen/@amzn/codewhisperer-streaming/node_modules/downlevel-dts/node_modules/typescript
```

After this change I expect it to continue to recurse into `src.gen/`, but the new ignore rules in `dependabot.yml` should auto-close PRs that modify it.

Reply:
> Without this, splitting off `src.gen` from the root repo results in a dependency error during packaging based on something in `src.gen/node_modules`.

That one sentence is most helpful for me, perhaps should be (first) in the code comment.

Does this add fragility to the build? Should we declare bankrupty and [fully separate](https://github.com/aws/aws-toolkit-vscode/pull/5667#pullrequestreview-2332061718) src.gen (whether by branch, subdir, or npm package)?

Reply:
> Does this add fragility to the build? Should we declare bankrupty and [fully separate](https://github.com/aws/aws-toolkit-vscode/pull/5667#pullrequestreview-2332061718) src.gen (whether by branch, subdir, or npm package)?

Obviously it adds some complexity but I wouldn't say its fragile. I don't think its worth the effort to separate src.gen/ if this works unless it causes problems later on.

Update: My initial interpretation of https://github.com/microsoft/vscode-vsce/issues/439 was incorrect. It seems that we don't need vsce's dependency checks at all since we are bundling the extension with webpack. So `--no-dependency` should be on by default for us and we don't need this custom checking logic that emulates the dependency checks.